src/core/controllers/oidc.go

```diff
@@ -244,6 +248,98 @@ func (oc *OIDCController) Callback() {
 	}
 }
 
+func (oc *OIDCController) RedirectLogout() {
+	sessionData := oc.GetSession(tokenKey)
+	ctx := oc.Ctx.Request.Context()
+	if sessionData == nil {
+		log.Error("OIDC session token not found.")
+		oc.SendInternalServerError(fmt.Errorf("OIDC session token not found"))
+		return
+	}
+	if err := oc.DestroySession(); err != nil {
+		log.Errorf("Error occurred in LogOut: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	oidcSettings, err := config.OIDCSetting(ctx)
+	if err != nil {
+		log.Errorf("Failed to get OIDC settings: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	if oidcSettings == nil {
+		log.Error("OIDC settings is missing.")
+		oc.SendInternalServerError(fmt.Errorf("OIDC settings is missing"))
+		return
+	}
+	if !oidcSettings.Logout {
+		oc.Controller.Redirect("/", http.StatusFound)
+		return
+	}
+	tk, ok := sessionData.([]byte)
+	if !ok {
+		log.Error("Invalid OIDC session data format.")
+		oc.SendInternalServerError(fmt.Errorf("invalid OIDC session data format"))
+		return
+	}
+	token := oidc.Token{}
+	if err := json.Unmarshal(tk, &token); err != nil {
+		log.Errorf("Error occurred in Unmarshal: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	if oidcSettings.LogoutOffline && token.RefreshToken != "" {
+		sessionType, err := getSessionType(token.RefreshToken)
```

Comment:
As we discussed, we should always try to revoke the token. 

I even think checking the session type can be optional, what will happen if the session it NOT offline?  Will the OIDC provider return any error?

Reply:
If the session type is not offline, there is no need to revoke it.